app/components/inputs/autoSuggestInput/useListOptions.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+import { DataListType } from "~/services/cms/components/StrapiAutoSuggestInput";
+import { DataListOptions } from "./getDataListOptions";
+
+const DATA_LIST_API_PATH = "/data-list-options";
+
+const useListOptions = (dataListType: DataListType["dataList"]) => {
+  const [dataListOptions, setDataListOptions] = useState<DataListOptions[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(`${DATA_LIST_API_PATH}/${dataListType}`);
+        const json = await response.json();
+        setDataListOptions(json);
+      } catch (error) {
+        setDataListOptions([]);
+      }
```

Comment:
we might need more check for the response, if you know that it's always 200 then we can ignore this.

```suggestion
      try {
        const response = await fetch(`${DATA_LIST_API_PATH}/${dataListType}`);
        if (response.ok) {
          const json = await response.json();
          setDataListOptions(json);
        }
      } catch (error) {
        setDataListOptions([]);
      }
```

Reply:
The changes are just a draft for now, I'll improve the code